### DISCUSSION TITLE
Passing parameters(current selected $this->user->id and not the auth()->user()->id) from view to createShowModal in livewire + laravel

### DISCUSSION THREAD

**Discussion by anis096-dev** [livewire/livewire#2791] [1. Help]
I want to pass the current selected user id to showmodal in livewire and in every attempt when I click to showmodal button I get this error: 
![image](https://user-images.githubusercontent.com/60985671/116556521-62ca4400-a8b2-11eb-9c41-a85435366f4e.png)
and if I delete {{$user->id}} the showmodal appears but when I submit the entered data I got the same error like:
![image](https://user-images.githubusercontent.com/60985671/116556860-c81e3500-a8b2-11eb-86f2-d08ac0d07843.png)
and this is my model data method:
`/**
     * The data for the model mapped
     * in this component.
     *
     * @return void
     */
    public function modelData()
    {  
        return [
            'user_id' => auth()->user()->id,
            'related_id' => $this->user->id,
            'treatment' => $this->treatment,
            'sub_treatment' => $this->sub_treatment,
            'status' => $this->status,
            ];
  }`
the create method:
`  /**
     * The create function.
     *
     * @return void
     */
    public function create()
    {
        $this->validate();
        Appointment::create($this->modelData());
        $this->modalFormVisible = false;
        $this->reset();
    }`
the create show modal is:
` /**
     * Shows the create modal
     *
     * @return void
     */
    public function createShowModal()
    {
        $this->resetValidation();
        $this->reset();
        $this->modalFormVisible = true;
    }`
and the render method like:
` public function render()
    {
        return view('livewire.user-appointments', [
            'data' => $this->read(),
        ]);
    }`

and the modal relationship is:
**App\User**
` public function appointments()
    {
        return $this->hasMany('App\Models\Appointment');
    }`
**App\Appointment**
`public function user()
    {
        return $this->belongsTo('App\Models\User');
    }`
any help please!

**Comment by Ahmadh1** [comment]
Well I think you should  write this
`public function render() { return view('livewire.user-appointments', [ 'data' => $this->read(), ]); }`
using  `mount()` method in component.
